kernel-modules/probe/collector_probe.c

```diff
@@ -227,17 +248,23 @@ static __always_inline int enter_probe(long id, struct sys_enter_args* ctx) {
     stack_ctx.id = id;
   }
 
-  sc_evt = get_syscall_info(id);
+#if defined(CAPTURE_SOCKETCALL)
+  if (id == __NR_socketcall) {
+    struct sys_enter_socketcall_args* socketcall_args = (struct sys_enter_socketcall_args*)ctx;
+    unsigned long socketcall_id = (unsigned long)socketcall_args->call;
+    mapped_id = convert_network_syscalls_by_id(socketcall_id);
+  }
+#endif
+
+  sc_evt = get_syscall_info(mapped_id);
   if (sc_evt == NULL || (sc_evt->flags & UF_USED) == 0) {
     return 0;
+  } else {
+    evt_type = sc_evt->enter_event_type;
+    drop_flags = sc_evt->flags;
   }
 
-  evt_type = sc_evt->enter_event_type;
-  drop_flags = sc_evt->flags;
-
-  // To satisfy some verifier requiremnts in later parts of the falco plumbing/fillers,
-  // it is necessary to copy the context onto the stack.
-  memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
+  syscall_to_enter_args(id, ctx, &stack_ctx);
```

Comment:
To reduce uncertainty / complexity on other platforms, we should only perform this copying on s390x. I think a simple ifdef guard here should be enough.

```suggestion
#ifdef __s390x__
  syscall_to_enter_args(id, ctx, &stack_ctx);
#else
  evt_type = sc_evt->enter_event_type;
  drop_flags = sc_evt->flags;
  // To satisfy some verifier requiremnts in later parts of the falco plumbing/fillers,
  // it is necessary to copy the context onto the stack.
  memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
#endif
```

Reply:
@Stringy Hmm... my concern here is that you copy the trace point specific format into a `struct sys_enter_args stack_ctx;` under those assumptions. This actually caused my a couple of BPF verifier issues e.g. if the tp format structure is shorter in length than `struct sys_enter_args`. In the worst case, you copy more memory than necessary (not sure why this seems to work on non-s390x architectures). Finally, that was the reason why I introduced those tp specific formats and assign the parameters directly.

Reply:
Yeah I completely see where you're coming from, but for the time being I think it would be best to isolate this new behaviour to s390x and leave the other architectures as they are, since they have been consistently working for some time. We can certainly revisit in a subsequent release, but with our intention to move to CO.RE, this may be sufficient until that point.

Reply:
resolved in https://github.com/kcrane/collector/commit/04a208802a9bec1159e32c31006af01b61de8bd5